### PR TITLE
Feature/archive dataset

### PR DESCRIPTION
**Pull request by chinook25**

### CODE REVIEW

File: src/main/webapp/resources/app/js/dataset/datasets.html [hunk 1]
```diff
@@ -2,37 +2,75 @@
...
+        <div class="cell">
+          <button ng-show="editMode.isUserOwner" tabindex="0" class="button small" ng-click="createDatasetDialog()">
+            <i class="fa fa-plus"></i>
+            Add new dataset
+          </button>
+          <button ng-disabled="numberOfDatasetsArchived === 0" ng-hide="showArchived"
+            class="button secondary small right ng-binding" ng-click="toggleShowArchived()"><i class="fa fa-eye"></i>
+            Show archived ({{numberOfDatasetsArchived}})
+          </button>
+          <button ng-disabled="false" ng-hide="!showArchived" class="button secondary small right ng-binding"
```
Comment (DanielReid) [inline src/main/webapp/resources/app/js/dataset/datasets.html]: ng-disabled="false" ???

File: src/test/java/org/drugis/trialverse/dataset/controller/DatasetControllerTest.java [hunk 2]
```diff
@@ -186,7 +187,8 @@ public void queryDatasetsRequestPathTurtleType() throws Exception {
 
   @Test
   public void queryDatasetsRequestPathJsonType() throws Exception {
-    List<Dataset> datasets = Arrays.asList(new Dataset("uri", john, "title", "description", "headVersion"));
+    String archivedOn = new Date().toString();
```
Comment (DanielReid) [inline src/test/java/org/drugis/trialverse/dataset/controller/DatasetControllerTest.java]: Test should be expanded to return list of both an archived and non-archived dataset

File: src/test/java/org/drugis/trialverse/dataset/controller/DatasetControllerTest.java [hunk 3]
```diff
@@ -362,4 +364,21 @@ public void testGetSpecificVersionInfo() throws Exception {
     verify(historyService).getVersionInfo(datasetUri, versionUri);
   }
 
+  @Test
+  public void testSetArchivedStatus() throws Exception {
+    String datasetUuid = "someDataset";
```
Comment (DanielReid) [inline src/test/java/org/drugis/trialverse/dataset/controller/DatasetControllerTest.java]: should also test the branch in the controller when john is not allowed to do this

File: src/test/java/org/drugis/trialverse/dataset/repository/DatasetReadRepositoryTest.java [hunk 4]
```diff
@@ -280,7 +278,7 @@ public void testExecuteVersionedQuery() throws URISyntaxException, IOException {
     String versionUuid = "http://myVersion.ninja";
     String acceptHeader = "confirm/deny";
     URI datasetUrl = new URI(Namespaces.DATASET_NAMESPACE + datasetUuid);
-    VersionMapping versionMapping = new VersionMapping(1, "http://whatever", "pietje@precies.gov", datasetUrl.toString());
+    VersionMapping versionMapping = new VersionMapping(1, "http://whatever", "pietje@precies.gov", datasetUrl.toString(), false, null);
```
Comment (DanielReid) [inline src/test/java/org/drugis/trialverse/dataset/repository/DatasetReadRepositoryTest.java]: Fix this with a constructor that doesn't need archived status imo

File: src/test/resources/schema.sql [hunk 5]
```diff
@@ -777,3 +777,9 @@ DROP TABLE toggledColumns;
 
 --changeset keijserj:80
 ALTER TABLE BenefitRiskStudyOutcomeInclusion ADD COLUMN baseline VARCHAR;
+
+--changeset keijserj:81
+ALTER TABLE VersionMapping ALTER COLUMN trialversedataseturl RENAME TO dataseturl;
```
Comment (DanielReid) [inline src/test/resources/schema.sql]: also undo refactor here

File: src/test/java/org/drugis/trialverse/dataset/repository/DatasetReadRepositoryTest.java (hunk 4 above)
Comment (chinook25) [reply src/test/java/org/drugis/trialverse/dataset/repository/DatasetReadRepositoryTest.java]: There is, for cases without ID. This fringe case only happens in very few places.

File: src/main/java/org/drugis/trialverse/util/JenaProperties.java [hunk 6]
```diff
@@ -23,6 +23,8 @@
   private static final String DC_DESCRIPTION = "http://purl.org/dc/terms/description";
   private static final String DC_CREATOR = "http://purl.org/dc/terms/creator";
   private static final String DC_DATE = "http://purl.org/dc/terms/date";
+  private static final String DRUGIS_ARCHIVED = "http://trials.drugis.org/ontology/archived";
```
Comment (DanielReid) [inline src/main/java/org/drugis/trialverse/util/JenaProperties.java]: I thought jena was no longer involved?

File: src/test/java/org/drugis/trialverse/dataset/repository/DatasetReadRepositoryTest.java [hunk 7]
```diff
@@ -100,10 +98,10 @@ public void testQueryDatasets() {
     Account account = new Account(1, "username", "firstName", "lastName", "foo@bar.com");
     String datasetLocation = "loc1";
     String versionKey = "version1";
-    VersionMapping versionMapping = new VersionMapping(1, datasetLocation, account.getEmail(), versionKey);
+    VersionMapping versionMapping = new VersionMapping(1, datasetLocation, account.getEmail(), versionKey, false, null);
```
Comment (DanielReid) [inline src/test/java/org/drugis/trialverse/dataset/repository/DatasetReadRepositoryTest.java]: these null arguments can be prevented with extra constructor for default 'null' archived state I think

File: src/test/java/org/drugis/trialverse/dataset/repository/DatasetWriteRepositoryTest.java [hunk 8]
```diff
@@ -49,6 +55,9 @@
   @Mock
   private WebConstants webconstants;
 
+  @Mock
+  private TriplestoreService triplestoreService;
```
Comment (DanielReid) [inline src/test/java/org/drugis/trialverse/dataset/repository/DatasetWriteRepositoryTest.java]: Is this still necessary?

File: src/test/resources/log4j.properties [hunk 9]
```diff
@@ -1,9 +1,9 @@
 # Root logger option
-log4j.rootLogger=ERROR, stdout, file
+log4j.rootLogger=DEBUG, stdout, file
```
Comment (DanielReid) [inline src/test/resources/log4j.properties]: is this still correct? Should keep not to change for love deploys maybe, logs are verbose enough as is there

File: src/test/java/org/drugis/trialverse/dataset/repository/DatasetReadRepositoryTest.java (hunk 7 above)
Comment (kzalite) [reply src/test/java/org/drugis/trialverse/dataset/repository/DatasetReadRepositoryTest.java]: The new constructor would only be used in the tests. Can live with the current implementation?

File: src/test/resources/log4j.properties (hunk 9 above)
Comment (kzalite) [reply src/test/resources/log4j.properties]: This configuration is used for tests. The one for deploys is on INFO.

**Pull request merged**